### PR TITLE
NextSecurity: Allow the installation directly from the USB stick

### PR DESCRIPTION
**Pull request by NethNick**
Added compressed ext4 image to itself (in a VFAT partition) 
Included install-nx.sh script to final image with support for multiple firmwares

### CODE REVIEW

**Comment by NethNick** [comment]
Installation directly from the USB stick with VFAT partition for firmware

File: builder/make.sh [hunk 1]
```diff
@@ -7,3 +7,25 @@
 
 # Start parallel build
 make -j $(nproc) V=sc world
+
+IMG=$(find /home/build/openwrt/bin -name nextsecurity\*-x86-64-generic-ext4-combined-efi.img.gz);
```
Comment (gsanchietti) [inline builder/make.sh:11]: Can you please avoid to look for only for x84_64 images?
We could just simply add a loop for all `nextsecurity\*-generic-ext4-combined-efi.img.gz` images

File: files/usr/sbin/install-nx.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:1]: Remember to always add the license header https://nethserver.github.io/nextsecurity/license_headers/.
```suggestion
#!/bin/bash
#
# Copyright (C) 2022 Nethesis S.r.l.
# SPDX-License-Identifier: GPL-2.0-only
#
```

File: files/usr/sbin/install-nx.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+              zcat $IMG| dd of=$T conv=notrunc
+           fi
+           umount /tmp/firmware
+           rmdir /tmp/firmware
+        else
+           if [ $M -eq 0 ]; then
+              echo -e "Multiple partitions find on target device, check it or use -f to force overwrite"
+           else
+              echo -e "Target partition in use, umount it first"
+           fi
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:56]: These are errors, the script should exit with non-zero code.
```suggestion
           if [ $M -eq 0 ]; then
              echo -e "Multiple partitions find on target device, check it or use -f to force overwrite"
           else
              echo -e "Target partition in use, umount it first"
           fi
           exit 1
```

File: files/usr/sbin/install-nx.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        fi
+else
+        echo -e "Target device not found"
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:59]: These are errors, the script should exit with non-zero code.
```suggestion
        echo -e "Target device not found"
        exit 1
```

File: files/usr/sbin/install-nx.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+if [ $# -eq 0 ]; then
+    echo -e "No arguments supplied, target device for installation needed\n$0 -t /dev/sdX [-s source]"    
+    exit 1
+fi
+while getopts "ft::s::" opt; do
+            case $opt in
+            (f) F=1 ;; #Force write
+            (s) S=${OPTARG} ;; #Source image
+            (t) T=${OPTARG} ;; #Target disk
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:10]: Please use variables with one chars only on limited scope, like inside loops.
Generally speaking variable names should explain the purpose.
Also, as convention, variables in upper case are usually constant.

So the above code should be something like:
```
            (f) force=1 ;; #Force write
            (s) source=${OPTARG} ;; #Source image
            (t) target=${OPTARG} ;; #Target disk
```

File: files/usr/sbin/install-nx.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+           if [ ! -f $IMG ]; then
+              echo "Firmware not found"
+              umount /tmp/firmware
+              rmdir /tmp/firmware
+              exit 1
+           else
+              zcat $IMG| dd of=$T conv=notrunc
+           fi
+           umount /tmp/firmware
+           rmdir /tmp/firmware
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:50]: You could avoid duplicated on line (43-44) by setting a trap at the top of the file:
```
trap "umount /tmp/firmware; rmdir /tmp/firmware" EXIT
```

**Review by gsanchietti — COMMENTED** [review]
Also remember to add the documentation inside [quickstart](https://github.com/NethServer/nextsecurity/blob/master/docs/quickstart.md).

File: files/usr/sbin/install-nx.sh [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+              read -r IMG
+              IMG=${B[$IMG]}
+           fi
+           if [ ! -f $IMG ]; then
+              echo "Firmware not found"
+              umount /tmp/firmware
+              rmdir /tmp/firmware
+              exit 1
+           else
+              zcat $IMG| dd of=$T conv=notrunc
```
Comment (gsanchietti) [inline files/usr/sbin/install-nx.sh:47]: The proposed change should speed up the write and avoid error if the image contains garbage at the end:
```suggestion
              zcat $IMG| dd of=$T  bs=64K iflag=fullblock conv=notrunc
```

**Review by NethNick — COMMENTED** [review]
Fullt agree with suggestions

File: builder/make.sh [hunk 8]
```diff
@@ -7,3 +7,25 @@
```
Comment (gsanchietti) [inline builder/make.sh:7]: Make sure to not execute other commands on failure:
```suggestion
set -e
```

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**